vms/platformvm/warp/README.md

```diff
@@ -0,0 +1,121 @@
+
+Note: the `blockchainID` is the hash of the transaction that created the blockchain on the Avalanche P-Chain. It serves as the unique identifier for the blockchain across the Avalanche Network.
+
+BitSet Signature:
+```
++-----------+----------+---------------------------+
+|   signers :   []byte |          4 + len(signers) |
++-----------+----------+---------------------------+
+| signature : [96]byte |                  96 bytes |
++-----------+----------+---------------------------+
+                       | 100 + size(signers) bytes |
+                       +---------------------------+
+```
+
+- `signers` encodes a BitSet of which validators' signatures are included
+- `signature` is an aggregated BLS Multi-Signature of the Unsigned Message
+
+BitSet Signatures are verified within the context of a specific P-Chain height. At a specific P-Chain height, the PlatformVM serves a canonically ordered validator set for the source subnet. The signers BitSet encodes which validators' signature was included. A 1 at index i in `signers` claims that a signature from the validator at index i in the canonical validator set is included in the aggregate signature.
+
+The BitSet tells the verifier which BLS Public Keys should be aggregated to verify the Warp Message.
+
+Signed Message:
+```
++------------------+------------------+-------------------------------------------------+
+| unsigned_message :  UnsignedMessage |                          size(unsigned_message) |
++------------------+------------------+-------------------------------------------------+
+|        signature :        Signature |                                 size(signature) |
++------------------+------------------+-------------------------------------------------+
+                                      |  size(unsigned_message) + size(signature) bytes |
+                                      +-------------------------------------------------+
+```
+
+## Sending an Avalanche Warp Message
+
+A Blockchain on Avalanche sends an Avalanche Warp Message by coming to agreement on the message that every validator should be willing to sign. As an example, the VM of a Blockchain may define that once a block is accepted, the VM should be willing to sign a message including the block hash in the payload to attest to any other Subnet that the block was accepted. The contents of the payload, how to aggregate the signature (VM-to-VM communication, off-chain relayer, etc.), is left to the VM.
+
+Once the validator set of a blockchain is willing to sign the message M, an aggregator performs the following process:
+
+1. Gather signatures of the message M from N validators (assume N validators meets the required threshold of stake on the destination chain)
+2. Aggregate the N signatures into a multi-signature
+3. Look up the canonical validator set at the P-Chain height where the message will be verified
```

Comment:
Same as above about defining "canonical"

Reply:
Defining once the first place where `canonical` is used